src/js/other-websites/add_input_icon.js

```diff
@@ -24,9 +24,29 @@ function addRelayMenuToPage(relayMenuWrapper, relayInPageMenu, relayIconBtn) {
 function positionRelayMenu() {
   const relayInPageMenu = document.querySelector(".fx-relay-menu-iframe");
   const relayIconBtn = document.querySelector(".fx-relay-menu-open");
+
+  if (!relayIconBtn) {
+    // BUGFIX: Catch erroneous/duplicate positionRelayMenu() calls before menu is open
+    return;
+  }
+
+  const relayInPageMenuIframe = document.querySelector(".fx-relay-menu-iframe");
   const newIconPosition = relayIconBtn.getBoundingClientRect();
-  relayInPageMenu.style.left = newIconPosition.x - 255 + "px";
-  relayInPageMenu.style.top = newIconPosition.top + 40 + "px";
+  const documentPosition = document.documentElement.getBoundingClientRect();
+  
+  // Calculate the "safe area" of add-on in-page menu. If there's not enough room to expand below the icon, it expands above. 
+  const positionMenuBelowIcon = ((((newIconPosition.top - documentPosition.top) - document.documentElement.scrollHeight) * -1) > 405);
```

Comment:
Comment added. I also made the "405" a variable so that there's more explanation in it too. `relayInPageMenuMaximumHeight`

Revised in 910f7c0ee528233571fd5878b68f59c3a9b067d6